st2common/st2common/hooks.py

```diff
@@ -166,8 +167,16 @@ class JSONErrorResponseHook(PecanHook):
 
     def on_error(self, state, e):
         error_msg = getattr(e, 'comment', str(e))
-        LOG.debug('API call failed: %s', error_msg)
-        LOG.debug(traceback.format_exc())
+        extra = {
+            'exception_class': e.__class__.__name__,
+            'exception_message': str(e),
+            'exception_data': e.__dict__
+        }
+
+        LOG.debug('API call failed: %s', error_msg, extra=extra)
+
+        if cfg.CONF.debug:
+            LOG.debug(traceback.format_exc())
```

Comment:
Answer : Yes.

https://github.com/stackforge/pecan/blob/master/pecan/core.py#L703

Not sure if it is safe to assume.